packages/types/src/async-utils.ts

```diff
@@ -0,0 +1,18 @@
+import { OnExecuteDoneEventPayload, OnExecuteDoneHookResult, OnExecuteDoneHookResultOnNextHook } from '@envelop/core';
+import isAsyncIterable from 'graphql/jsutils/isAsyncIterable.js';
+
+export function handleMaybeStream(
```

Comment:
`handleStreamOrSingleExecutionResult` ?

Also, we can add a jsdoc on why it is useful - with a short usage example?